### DISCUSSION TITLE
How to get a list to all markdown pages?

### DISCUSSION THREAD

**Discussion by abd3lraouf** [varabyte/kobweb#443] [Q&A] [answered]
Do I need a worker to it? the data should be static already so I'm thinking of some API that will be used to get them

Basically, I need to create a blog of articles defined in markdown, and the index will contain a list to all of them, I will get their title, image, author and date

**Comment by abd3lraouf** [comment]
I think we can make use of the ksp plugin to generate such thing

**Comment by bitspittle** [comment, marked as answer]
Apologies for missing your question until now!

I saw you found the relevant bug, which is good! But the current workaround is to create a Gradle task. You can look at what I'm doing here: https://github.com/bitspittle/bitspittle.dev/blob/7ec7a4a9e89ee2b41974b0843a90688bb0a9605c/site/build.gradle.kts#L104

which I use for generating this page here: https://bitspittle.dev/blog/

**Comment by abd3lraouf** [comment]
Thank you so much @bitspittle I've already resolved it and using at least 4 gradle tasks to generate the following:
- category indexes
- tags page
- tags/tagName/index
- MarkdownData object 

and it was a great fun :) thank you for pointing me to the right direction

**Reply by bitspittle** [reply]
You're welcome. FYI working on a feature in 0.15.5 (going out in the next few days) that should make this slightly easier. Keep an eye out for it because I'll be sure to mention it in the release notes.

Although if you're already working, honestly that's what matters most and I'm glad you had fun with it!
